numismatic/feeds/base.py

```diff
@@ -95,6 +107,30 @@ def write_to_file(batch):
              .filter(len)
              .sink(write_to_file)
              )
+
+        ws = await self._connect(wss_url)
+        channel_info = {'channel': channel}
+        return ws, channel_info
+
+    @classmethod
+    async def _unsubscribe(cls, ws, symbol):
+        return True
+
+    async def listen(self, symbol, channel=None, wss_url=None):
+        symbol = symbol.upper()
+        ws, channel_info = await self._subscribe(symbol,  channel, wss_url)
+        while True:
```

Comment:
I hate infinite loops. Seems exremely "hacky". Is there any way we can get around this? I understand the concept of running all the time, just wondering if there is a way to gracefully exit this.

Reply:
From the asyncio documentation it looks like handling the CancelledError exception is the way this is done.